### DISCUSSION TITLE
Fabric JS based video editor

### DISCUSSION THREAD

**Discussion by hihilisk** [fabricjs/fabric.js#8509] [General]
Hi! 
I wanna build an online video editor based on fabricjs. I’ve created a class Video and I extended fabric in my local project… I can add videos on canvas, I can manipulate, add filters, and other stuff like for images. But! I have some troubles with performance. I’m using canvas.getElement().captureStream() and it seems it will not work for a lot of users (slow laptops, slow PC, etc…). It takes almost all resources of my laptop when captureStream() works. FPS about 20-25, in video i get frame loss. 

Do you guys have an Idea how to improve my approach? 

 For auto re-render i use custom function:

```JS
// fabric.Canvas
handlePlayStart() {
  const canvas = this;

  canvas.autoReRenderId = setInterval(() => {
    canvas.requestRenderAll();
  }, 15);
},
```

**Comment by hihilisk** [comment]
@kangax @davidjrice @maxim @seanhussey

**Comment by ShaMan123** [comment]
I suggest you provide an online repro using codesandbox or something.
You should do an infinite call to requestRenderAll as follows:
```js
const run = () => fabric.util.requestAnimFrame(() => {
   canvas.renderAll();
   run();
})
run();
```

OR

```js
canvas.on('after:render', () => canvas.requestRenderAll());
canvas.requestRenderAll();

```
